modules/network/src/main/java/org/apache/ignite/network/internal/direct/DirectMarshallingUtils.java

```diff
@@ -18,31 +18,33 @@
 package org.apache.ignite.network.internal.direct;
 
 import java.nio.ByteBuffer;
+import org.apache.ignite.network.internal.direct.stream.DirectByteBufferStream;
 
 /**
  * Direct marshalling utils.
  */
 public class DirectMarshallingUtils {
     /**
-     * Reads a direct message type from a byte buffer.
+     * Reads a {@code short} from a byte buffer in an order defined by the {@link DirectByteBufferStream}
+     * implementation.
      *
      * @param buffer Byte buffer.
      * @return Direct message type.
      */
-    public static short getMessageType(ByteBuffer buffer) {
+    public static short getShort(ByteBuffer buffer) {
         byte b0 = buffer.get();
```

Comment:
I would compress this and assert during write (not only) that type cannot be more then 0x3FFF. We should have compressed shorts somewhere already, don't we?

Reply:
Unfortunately, the `DirectByteBufferStream` does not compress shorts: https://github.com/apache/ignite-3/blob/main/modules/network/src/main/java/org/apache/ignite/network/internal/direct/stream/DirectByteBufferStreamImplV1.java#L199

Reply:
Can we get back the name "getMessageType" and implement compression here?

Reply:
but we will need to implement compression on the writing side as well, won't we? And I will have to write tests for that. Can we do it as a separate task, I'm afraid this PR is already too big?